src/main/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFBlockMergingIterator.java

```diff
@@ -0,0 +1,14 @@
+package org.broadinstitute.hellbender.utils.variant.writers;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.utils.iterators.PushToPullIterator;
+
+import java.util.*;
+
+public class GVCFBlockMergingIterator extends PushToPullIterator<VariantContext> {
```

Comment:
I would add a comment here making explicit the fact that you are reusing the PushToPullIterator which talks about downsampling in its comments and indeed appears to have that hardwired into its behavior. Or I would make the commenting on PushToPullIterator more generic (referring to downsampling as an example) to avoid confusion.

Reply:
I've reworked the javadoc in #5311 to address this.